rust/agama-lib/src/logs.rs

```diff
@@ -0,0 +1,336 @@
+// Copyright (c) [2024] SUSE LLC
+//
+// All Rights Reserved.
+//
+// This program is free software; you can redistribute it and/or modify it
+// under the terms of the GNU General Public License as published by the Free
+// Software Foundation; either version 2 of the License, or (at your option)
+// any later version.
+//
+// This program is distributed in the hope that it will be useful, but WITHOUT
+// ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+// FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+// more details.
+//
+// You should have received a copy of the GNU General Public License along
+// with this program; if not, contact SUSE LLC.
+//
+// To contact SUSE LLC about this file by physical or electronic mail, you may
+// find current contact information at www.suse.com.
+
+use crate::error::ServiceError;
+use fs_extra::copy_items;
+use fs_extra::dir::CopyOptions;
+use serde::Serialize;
+use std::fs;
+use std::fs::File;
+use std::io;
+use std::io::Write;
+use std::os::unix::fs::PermissionsExt;
+use std::path::{Path, PathBuf};
+use std::process::Command;
+use tempfile::TempDir;
+use utoipa::ToSchema;
+
+const DEFAULT_COMMANDS: [(&str, &str); 3] = [
+    // (<command to be executed>, <file name used for storing result of the command>)
+    ("journalctl -u agama", "agama"),
+    ("journalctl -u agama-auto", "agama-auto"),
+    ("journalctl --dmesg", "dmesg"),
+];
+
+const DEFAULT_PATHS: [&str; 14] = [
+    // logs
+    "/var/log/YaST2",
+    "/var/log/zypper.log",
+    "/var/log/zypper/history*",
+    "/var/log/zypper/pk_backend_zypp",
+    "/var/log/pbl.log",
+    "/var/log/linuxrc.log",
+    "/var/log/wickedd.log",
+    "/var/log/NetworkManager",
+    "/var/log/messages",
+    "/var/log/boot.msg",
+    "/var/log/udev.log",
+    // config
+    "/etc/install.inf",
+    "/etc/os-release",
+    "/linuxrc.config",
+];
+
+const DEFAULT_RESULT: &str = "/run/agama/agama-logs";
+// what compression is used by default:
+// (<compression as distinguished by tar>, <an extension for resulting archive>)
+pub const DEFAULT_COMPRESSION: (&str, &str) = ("gzip", "tar.gz");
+const TMP_DIR_PREFIX: &str = "agama-logs.";
+
+/// Configurable parameters of the "agama logs" which can be
+/// set by user when calling a (sub)command
+pub struct LogOptions {
+    paths: Vec<String>,
```

Comment:
I wonder if the list of paths and commands is part of the options. IMHO, that should not be the caller's responsibility.

Reply:
Well original idea was to prepare a way how to make possible to add a reference to a file as parameter. This doesn't look as promising idea anymore. I'll change that.